File: codeforces/gym/2019-2020 ACM-ICPC Latin American Regional Programming Contest/M/m.cpp
#include<iostream>

using namespace std;

int main(int argc, char const *argv[])
{
    int n, x;
    cin >> n >> x;
    int a[n];
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    int ans = 0, count = 1;
    for (int i = 0; i < n - 1; i++) {
        if (a[i + 1] - a[i] <= x) {
            count++;
        } else {
            ans = max(ans, count);
            count = 1;
        }
    }
    ans = max(ans, count);
    cout << ans << endl;
    return 0;
}

File: codeforces/gym/2019-2020 ACM-ICPC Latin American Regional Programming Contest/M/m.py
n, x = [int(i) for i in input().split()]

a = [int(i) for i in input().split()]

ans, count = 0, 1

for i in range(n - 1):
    if a[i + 1] - a[i] <= x:
        count += 1
    else:
        ans = max(ans, count)
        count = 1
ans = max(ans, count)
print(ans)